segmenters/nlp/PDFPlumberSegmenter/__init__.py

```diff
@@ -0,0 +1,93 @@
+import io
+from typing import Dict, List
+
+import numpy as np
+
+from jina.executors.decorators import single
+from jina.executors.segmenters import BaseSegmenter
+
+class PDFPlumberSegmenter(BaseSegmenter):
+    """
+    :class:`PDFPlumberSegmenter` Extracts data (text and images) from PDF files.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    @single(slice_nargs=3)
+    def segment(self, uri: str, buffer: bytes, mime_type: str, *args, **kwargs) -> List[Dict]:
+        """
+        Segements PDF files. Extracts data from them.
+
+        Checks if the input is a string of the filename,
+        or if it's the file in bytes.
+        It will then extract the data from the file, creating a list for images,
+        and text.
+
+        :param uri: File name of PDF
+        :type uri: str
+        :param buffer: PDF file in bytes
+        :type buffer: bytes
+        :param mime_type: the type of data
+        :returns: A list of documents with the extracted data
+        :rtype: List[Dict]
+        """
+        chunks = []
```

Comment:
If we need this, we should add a FliterQL driver before this executor, but this should not be part here in my opinion

Reply:
You mean `chunks = []`? Other segments also have this. Or you mean the mimetype?

Reply:
Ah yes, we can add a driver. But this is easier to use. And I plan to add some warning information here to suggest users to use driver to filter only pdf data.

Reply:
I am not convinced with this. This should not be this executors responsability I think

Reply:
How can we solve this situation? I have 3 segments working sequentially, 1 text, 1 image, and 1 pdf. If I use FliterQL at first for the text segmenter, how can I keep the query data of images and pdf?

Reply:
Why do u have these Sequentially?

Reply:
I tried to use them in parallel but will get issues. So I use the compound to stack these 3 together.

Reply:
ok keep it like this, but is weird

Reply:
Yeah, true. I also don't like it. 😂 
I came up with 2 ways for this case:

- Use the 3 segmenters in parallel and each equipped with a filter driver. This is more aligned with our design. But I got very strange issues and cannot reproduce each time. I will try to reproduce it later and maybe open an issue.
- Use the 3 segmenters in sequential, and add this check inside the segmenters.